### DISCUSSION TITLE
Phase shift between timers

### DISCUSSION THREAD

**Discussion by tjc2k4** [Dlloydev/ESP32-ESP32S2-AnalogWrite#32] [General]
I'd like to create a lot of synchronized PWM outputs where they are phase shifted a specific amount relative to each other. I'm able to do it based on extending the 3phase sample, but there's some shift amount between the channels that use different timers. I see how the sample corrects for it but I have some questions about it
Where does that delay/shift come from? 
And is it be constant on every ESP32 or would it need to be adjusted on each? 

Thanks!

**Comment by Dlloydev** [comment]
There are 8 channel pairs (0-1, 2-3, 4-5, etc). There is no phase shift for the signals in each pair because they use the same timer. However, there's a delay/shift that occurs between each channel pair due to software overhead and when the timers are actually started.  This can be minimized somewhat by using the pause and resume functions, but can't be eliminated. It can be corrected as shown in the 3-phase examples, but each software (library) revision would produce different results.

Only the MCPWM peripheral of the ESP32 has hardware sync capability. At higher frequencies, this might be the only solution, depending on your application.
